tools/ci_build/build.py

```diff
@@ -1325,10 +1327,13 @@ def generate_build_tree(
     if args.use_snpe:
         cmake_args += ["-Donnxruntime_USE_SNPE=ON"]
 
+    # minor note: the below condition basically just means an apple framework build
+    # and it can cover platforms including macosx/iphoneos/iphonesimulator/maccatalyst.
     if args.build_apple_framework or args.ios:
```

Comment:
What's the relationship between build_apple_framework and mac_catalyst? 

If I only specify mac_catalyst neither build_apple_framework or ios would be set, and it wouldn't hit the below check on what cmake_generator is valid. Is that correct and the cmake_generator doesn't matter if I do something simple like `build.py --mac_catalyst`? Or if I am expected/required to specify other params with `--mac_catalyst` the help for that argument should call it out.

Reply:
thanks for pointing out this. yeah I wrongly assumed the argument would always specify as a combination of `--mac_catalyst` + `build_apple_framework`. 
perhaps I will add changes to note in this script that the `--mac_catalyst` argument would only be specified when we are building static apple framework and the behavior other than this combination is unsure for now. 

originally we put a condition here - `args.build_apple_framework or args.ios` is mainly for eliminating an extra argument for `--macosx` (given we'll always specify `--build_apple_framework` for macos builds so checking it should be enough.) and now since we have an additional catalyst arch, maybe it'll be worth to add an additional argument like `--macos Catalyst/MacOSX` with the choices to specify variant for Catalyst, macosx builds.